libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -502,7 +530,34 @@ bool ExpressionCompiler::visit(BinaryOperation const& _binaryOperation)
 	CompilerContext::LocationSetter locationSetter(m_context, _binaryOperation);
 	Expression const& leftExpression = _binaryOperation.leftExpression();
 	Expression const& rightExpression = _binaryOperation.rightExpression();
-	solAssert(!!_binaryOperation.annotation().commonType, "");
+	FunctionDefinition const* function = *_binaryOperation.annotation().userDefinedFunction;
+	if (function)
+	{
+		solAssert(function->isFree());
+
+		FunctionType const* functionType = _binaryOperation.userDefinedFunctionType();
+		solAssert(functionType);
+		solAssert(functionType->parameterTypes().size() == 2);
+		solAssert(functionType->returnParameterTypes().size() == 1);
+		solAssert(functionType->kind() == FunctionType::Kind::Internal);
+
+		evmasm::AssemblyItem returnLabel = m_context.pushNewTag();
+		acceptAndConvert(leftExpression, *functionType->parameterTypes()[0]);
+		acceptAndConvert(rightExpression, *functionType->parameterTypes()[1]);
```

Comment:
```suggestion
		acceptAndConvert(leftExpression, *functionType->parameterTypes()[0], true);
		acceptAndConvert(rightExpression, *functionType->parameterTypes()[1], true);
```

Reply:
Left as is. See https://github.com/ethereum/solidity/pull/13790#discussion_r1113626349.